components/plurals/src/lib.rs

```diff
@@ -304,28 +304,46 @@ impl PluralRules {
     ) -> Result<Self, PluralRulesError>
     where
         D: ResourceProvider<CardinalV1Marker> + ResourceProvider<OrdinalV1Marker> + ?Sized,
+    {
+        match rule_type {
+            PluralRuleType::Cardinal => Self::try_new_cardinal(locale, data_provider),
+            PluralRuleType::Ordinal => Self::try_new_ordinal(locale, data_provider),
+        }
+    }
+
+    pub fn try_new_cardinal<T: Into<Locale>, D>(
+        locale: T,
+        data_provider: &D,
+    ) -> Result<Self, PluralRulesError>
+    where
+        D: ResourceProvider<CardinalV1Marker> + ?Sized,
     {
         let locale = locale.into();
-        let rules = match rule_type {
-            PluralRuleType::Cardinal => ResourceProvider::<CardinalV1Marker>::load_resource(
-                data_provider,
-                &DataRequest {
-                    options: locale.clone().into(),
-                    metadata: Default::default(),
-                },
-            )?
+        let rules = data_provider
+            .load_resource(&DataRequest {
+                options: locale.clone().into(),
+                metadata: Default::default(),
+            })?
             .take_payload()?
-            .cast(),
-            PluralRuleType::Ordinal => ResourceProvider::<OrdinalV1Marker>::load_resource(
-                data_provider,
-                &DataRequest {
-                    options: locale.clone().into(),
-                    metadata: Default::default(),
-                },
-            )?
+            .cast();
```

Comment:
question: hmm, after this is PluralRulesV1Marker only internal to PluralRules? shoudl we be trying to get rid of it completely?

Reply:
We could perhaps make it internal to PluralRules.  But, we do need a common marker type to avoid PluralRules having to be parameterized in a user-visible way.

Reply:
Sounds good.

Reply:
Let's document it as internal at least

Reply:
Hmm.  `PluralRulesV1Marker` is auto-generated by the macro, so I can't easily make it non-pub.

Personally I don't see a problem with `pub struct PluralRulesV1Marker` because it directly corresponds to `pub struct PluralRulesV1`.  It's all inside the `provider` module anyway.

Reply:
I made it private.

Reply:
I had to make it public again because it is used by `PluralsProvider`.  Follow-up: #1575